### ISSUE TITLE
Question about moving tdewolff/parse into tdewolff/minify

### ISSUE THREAD

**Issue by karelorigin** [tdewolff/minify#416]
Hi! I recently noticed the original parse repository was moved here, may I ask why? Doesn't it seem more convenient to have a separate repository for the underlying parsers?

I also forked the repository some days before it happened and added some additional features such as Walking the AST tree and converting it back to valid JS. I believe this would be a useful addition to the JS parser. You can find it here if you're interested: [karelorigin/parse](https://github.com/karelorigin/parse)

**Comment by tdewolff** [comment]
There have been many requests in the past to merge the libraries, and for me as a maintainer it seems easier as well. The idea was to have separate parsing libraries that could be used for other goals, but in the end it's only ever being used for `minify`. Every release of `parse` triggers a release of `minify`, which was cumbersome. I figured this way maintenance would be easier, and people could still use the parse library separately, just by changing the import path to `github.com/tdewolff/minify/parse`. Could you please elaborate why you think they should be separate? I might reconsider...

Your work seems a useful addition to the library, I would be glad to accept a PR to incorporate it, thanks!

**Comment by karelorigin** [comment]
I believe other projects importing the library might end up (indirectly) importing all dependencies `minify` itself relies on. Contributions to the `parse` library also seem easier when it's not encapsulated by another project. I also think you might be underestimating the usefulness of the `parse` library, I found myself in a position where there weren't any alternatives I could use. `parse/js` pretty much did everything I wanted it to except for the functionality I ended up adding myself. The biggest issue, as far as I can tell, appears to be incorporating new changes from `parser` into `minify`. I believe this process was automated, right? 

In my opinion, it just makes more sense to keep unrelated stuff separate to keep things clear and modular. Creating a PR would mean moving my changes from `parse` to `minify` as `parse` is now archives, so let me know what you think 😅
